### PR TITLE
Some stuff

### PR DESCRIPTION
**Pull request by tombigel**
Ignore all the .dist files etc of course... this is why they need to be in a different ignored folder...

see commit descriptions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "demo:build:parallax": "cd docs/demo && rollup -c parallax.config.js && cd -",
     "demo:build:background": "cd docs/demo && rollup -c background.config.js && cd -",
     "demo:build": "npm run demo:build:single && npm run demo:build:layers && npm run demo:build:parallax && npm run demo:build:background",
-    "demo": "npm run build && npm run copy && npm run demo:build"
+    "demo": "npm run build && npm run copy && npm run demo:build",
+    "demo:start": "npm run demo && http-server ./docs/demo"
```
Comment (ydaniv) [inline package.json:16]: 👍

File: src/Scroll.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ export default class Scroll extends Two5 {
     constructor (config = {}) {
         super(config);
 
+        this.config.root = this.config.root || window;
```
Comment (ydaniv) [inline src/Scroll.js:24]: `scrollTo`, `scrollY`, and `pageYOffset` are only properties of `window`, not of `Element`.
Also, can you think of a use case that will require to override root?

File: src/Tilt.js [hunk 3]
```diff
@@ -72,16 +72,16 @@ export default class Tilt extends Two5 {
      */
     setupEvents () {
         // attempt usage of DeviceOrientation event
-        const gyroscoeHandler = getGyroscope({
+        const gyroscopeHandler = getGyroscope({
```
Comment (ydaniv) [inline src/Tilt.js:75]: 👍

File: src/Tilt.js [hunk 4]
```diff
@@ -144,12 +144,12 @@ export default class Tilt extends Two5 {
...
- * @property {number} gyroscopeSamples
- * @property {Element} mouseTarget
- * @property {Element} layersContainer
+ * @property {boolean} [animationActive] whether to animate effect progress.
+ * @property {number} [animationFriction] between 0 to 1, amount of friction effect in the animation. 1 being no movement and 0 as no friction. Defaults to 0.4.
+ * @property {number} [maxBeta]
+ * @property {number} [maxGamma]
+ * @property {number} [gyroscopeSamples]
+ * @property {Element} [mouseTarget]
+ * @property {Element} [layersContainer]
```
Comment (ydaniv) [inline src/Tilt.js:153]: 👍 great

File: src/Two5.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { defaultTo, lerp } from './utilities.js';
 
 const ticker = {
-    pool: [],
+    pool: new Set(),
```
Comment (ydaniv) [inline src/Two5.js:4]: 👍 liked this one, made `add()` and `remove()` more succinct

File: src/Two5.js [hunk 6]
```diff
@@ -215,8 +208,8 @@ export default class Two5 {
...
 /**
  * @typedef {Object} two5Config
- * @property {boolean} animationActive whether to animate effect progress.
- * @property {number} animationFriction from 0 to 1, amount of friction effect in the animation. 1 being no movement and 0 as no friction. Defaults to 0.4.
- * @property {boolean} velocityActive whether to calculate velocity with progress.
- * @property {number} velocityMax max possible value for velocity. Velocity value will be normalized according to this number, so it is kept between 0 and 1. Defaults to 1.
+ * @property {boolean} [animationActive] whether to animate effect progress.
+ * @property {number} [animationFriction] from 0 to 1, amount of friction effect in the animation. 1 being no movement and 0 as no friction. Defaults to 0.4.
+ * @property {boolean} [velocityActive] whether to calculate velocity with progress.
+ * @property {number} [velocityMax] max possible value for velocity. Velocity value will be normalized according to this number, so it is kept between 0 and 1. Defaults to 1.
```
Comment (ydaniv) [inline src/Two5.js:214]: 👍

File: src/effects/scroll-controller.js [hunk 7]
```diff
@@ -102,34 +105,34 @@ export function getEffect (config) {
     /*
      * Setup Smooth Scroll technique
      */
-    if (container) {
+    if (container && body.contains(container)) {
```
Comment (ydaniv) [inline src/effects/scroll-controller.js:108]: I think this is really not necessary. I'd even argue it's wrong to assert this. Can you explain your motivation?

File: src/effects/scroll-controller.js [hunk 8]
```diff
@@ -102,34 +105,34 @@ export function getEffect (config) {
...
         else {
-            window.document.body.style.height = `${totalHeight}px`;
+            body.style.height = `${totalHeight}px`;
         }
 
         /*
          * Setup wrapper element and reset progress.
          */
-        if (_config.wrapper) {
+        if (wrapper && wrapper.contains(container) && body.contains(wrapper)) {
```
Comment (ydaniv) [inline src/effects/scroll-controller.js:124]: Again, I believe it's wrong to add these checks, what is your motivation?

File: src/Scroll.js (hunk 2 above)
Comment (tombigel) [reply src/Scroll.js:24]: **scrollTo, scrollY, and pageYOffset are only properties of window, not of Element.**
yeah.. I meant scroll top, will revise. pageYOffset can be retired, its IE9 related prop
**can you think of a use case that will require to override root?**
yes, render a site not in its own window, like add panel.

File: src/effects/scroll-controller.js (hunk 7 above)
Comment (tombigel) [reply src/effects/scroll-controller.js:108]: you code accepts hierarchy scroll -> optional wrapper -> container. it needs to be enforced or at least throw an error (developer UX...)

File: src/effects/scroll-controller.js (hunk 8 above)
Comment (tombigel) [reply src/effects/scroll-controller.js:124]: see previous comment

File: package.json (hunk 1 above)
Comment (tombigel) [reply package.json:16]: donno, i opened the demos and they worked.
but I agree about the watch, we should have  it

File: src/Scroll.js (hunk 2 above)
Comment (tombigel) [reply src/Scroll.js:24]: pushed fix for scrollTop

Comment (ydaniv) [reply src/Scroll.js:24]: Isn't it IE11?

Add panel is bad example, but I can think of a horizontally scrolled strip that we want to control separately. Does that make sense?

Comment (tombigel) [reply src/Scroll.js:24]: makes sense. think of a gallery that uses scroll snaps, think of a live site preview without an iframe... it could be cool

Comment (ydaniv) [reply src/Scroll.js:24]: simple snaps for gallery can be achieved with CSS, but OK

File: src/effects/scroll-controller.js [hunk 9]
```diff
@@ -102,34 +105,34 @@ export function getEffect (config) {
...
                 height: '100vh',
                 overflow: 'hidden'
             });
 
-            // get current scroll position
-            let x = window.scrollX || window.pageXOffset;
-            let y = window.scrollY || window.pageYOffset;
+            // get current scroll position (support window, element and window in IE)
+            let x = root.pageXOffset || root.scrollX || root.scrollLeft || 0;
+            let y = root.pageYOffset || root.scrollY || root.scrollTop || 0;
```
Comment (ydaniv) [inline src/effects/scroll-controller.js:135]: scrollY is the future API, it should be the first option,not the fallback

File: src/effects/scroll-controller.js (hunk 7 above)
Comment (tombigel) [reply src/effects/scroll-controller.js]: I'm not sure your'e right, but removed.

File: src/effects/scroll-controller.js (hunk 8 above)
Comment (tombigel) [reply src/effects/scroll-controller.js]: removed

File: src/effects/scroll-controller.js (hunk 9 above)
Comment (tombigel) [reply src/effects/scroll-controller.js]: fixed

**Pull request merged**